File: CustomerApp/ConsulApp/Models/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ConsulApp.Models
{
    public class Customer
    {
        [Key]
        public int Id { get; set; }
       
        [MaxLength(50)]
        public string Name { get; set; }
        [MaxLength(20)]
        public string Country { get; set; }
        [MaxLength(50)]
        public string State { get; set; }
        [MaxLength(100)]
        public string Address { get; set; }
        [Required]
        [MaxLength(50)]
        [Index(IsUnique = true)]

        public string Email { get; set; }

        [MaxLength(10)]
        public string PAN { get; set; }
        [MaxLength(10)]
        public string Phone { get; set; }
        public DateTime DOB { get; set; }
        [MaxLength(10)]
        public string PrimaryContact { get; set; }
        [Required]
        [MaxLength(20)]
        public string Password { get; set; }
        [MaxLength(10)]
        public string Role { get; set; }
    }
}

File: AngularApp/src/app/Models/Customer.ts
import { AbstractControl, FormBuilder, FormControl, FormGroup, Validators } from "@angular/forms";

export class Customer{
    Id:number=0;
    Name:string="";
    Country:string="";
    State:string="";
    Address:string="";
    Email:string="";
    PAN:string="";
    Phone:string="";
    PrimaryContact:string="";
    Password:string="";
    DOB:Date = new Date;
    Role:string = "";
    public formCustomerGroup : FormGroup;
constructor(){
    var _builder = new FormBuilder();

  
    this.formCustomerGroup = _builder.group({
        userNameControl: new FormControl('',Validators.compose([Validators.required])),
        userPhoneContrl: new FormControl('',Validators.compose([Validators.required,Validators.pattern("[0-9 ]{10}")])),
        userEmailControl: new FormControl('',Validators.compose([Validators.required,Validators.email])),
        userReqControl: new FormControl('',Validators.compose([Validators.required])),
        passControl : new FormControl('',Validators.required),
        confirmControl : new FormControl('',Validators.required),
        panControl : new FormControl('',Validators.compose([Validators.required,Validators.pattern("[A-Z]{5}[0-9]{4}[A-Z]{1}")])),
        stateControl : new FormControl('',Validators.required),
        countryCntrol : new FormControl('',Validators.required),
        dateControl : new FormControl('',Validators.required),
        addressControl :new FormControl('',Validators.required)
    },
    {
    validators : this.mustMatch('passControl','confirmControl')
    }
    
    );
    
}
mustMatch(p:any,cp:any){
    return (formGroup:FormGroup) =>{
        const passControl = formGroup.controls[p];
        const confirmControl = formGroup.controls[cp];
        if(confirmControl.errors && !confirmControl.errors['mustMatch']){
            return;
        }
        if(passControl.value!==confirmControl.value)
        {
            confirmControl.setErrors({mustMatch:true});
        }
      
        else{
            confirmControl.setErrors(null);
        }
    };

}
   
}